src/upload-artifact.ts

```diff
@@ -21,16 +21,24 @@ async function run(): Promise<void> {
 
       const artifactClient = create()
       const options: UploadOptions = {
-        continueOnError: true
+        continueOnError: false
       }
-      await artifactClient.uploadArtifact(
+      const uploadResponse = await artifactClient.uploadArtifact(
         name || getDefaultArtifactName(),
         searchResult.filesToUpload,
         searchResult.rootDirectory,
         options
       )
 
-      core.info('Artifact upload has finished successfully!')
+      if (uploadResponse.failedItems.length > 0) {
+        core.setFailed(
```

Comment:
I double checked my code in `@actions/artifact` and it's up to us to call `core.setFailed` it something doesn't get uploaded. If an error is encountered, it just calls [core.error](https://github.com/actions/toolkit/blob/a67b91ea15598bea0bbf639ef4e91916feab0c69/packages/artifact/src/internal/upload-http-client.ts#L168) which doesn't actually fail the step.